### ISSUE TITLE
Why extreme differences in the ages of a couple?

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/taxdata#18]
I don't understand why there are some couples (that is, filing units in `puf.csv` with `MARS==2`) with very large age differences.  Most (about 90 percent of the sample weight of all couples) have the same age as shown by the following results:

```
$ awk -F, 'NR>1&&$11==2{n[$2-$1]+=$102;w+=$102}END{for(ad in n)print ad,100*n[ad]/w}' puf.csv | awk '$1==0'
0 90.2593
```

And even more (over 97 percent of couples) have a spouse-minus-head age difference in the [-9,+9] range:

```
$ awk -F, 'NR>1&&$11==2{n[$2-$1]+=$102;w+=$102}END{for(ad in n)print ad,100*n[ad]/w}' puf.csv | sort -g | awk '$1>=-9&&$1<=9{p+=$2}END{print p}'
97.5775
```

But I don't understand why the logic generates a few extremely large age differences as seen in these results:

```
$ awk -F, 'NR>1&&$11==2{n[$2-$1]+=$102;w+=$102}END{for(ad in n)print ad,100*n[ad]/w}' puf.csv | sort -g > ad
$ head ad
-84 0.0372232
-79 0.0712607
-78 0.00594574
-77 0.0163171
-76 0.0220896
-75 0.0283592
-74 0.0201275
-73 0.00479531
-72 0.0440797
-71 0.0132917
$ tail ad
27 0.0125647
28 0.000265776
29 0.000887838
30 0.00696035
33 0.000555454
35 0.000655087
36 3.40456e-05
38 0.000389501
42 0.0031806
45 0.0077123
$ 
```

I'm aware of the concept of a "trophy wife", but filing units where the age of the spouse is more than seventy years less than the age of the head seem a little extreme.

Does anybody understand why we are generating these few extreme age differences?

@Amy-Xu @MattHJensen

**Comment by martinholmer** [comment]
The extreme age differences between head and spouse problem identified in taxdata issue #18, has been found to be caused by the SAS scripts that generated the `cps-matched-puf.csv` file as described in #87.  The merged pull request #87 is a temporary work-around this problem.  As described in #87, the long-run plan is to fix the SAS scripts, which will mean the `age_spouse` variable in the  `cps-matched-puf.csv` file will be more sensible.  Once that happens, the `puf_data/finalprep.py` script will be revised to rely more on the higher-quality `age_spouse` data.

@MattHJensen @feenberg @Amy-Xu @andersonfrailey

**Comment by martinholmer** [comment]
taxdata issue #18 has been resolved by pull request #100 as described in issue #92.
